cbits/dns.c

```diff
@@ -0,0 +1,84 @@
+#include <winsock2.h>
+#include <iphlpapi.h>
+#include <string.h>
+#include <windows.h>
+#include <string.h>
+#include "dns.h"
+
+#define MALLOC(x) HeapAlloc(GetProcessHeap(), 0, (x))
+#define FREE(x) HeapFree(GetProcessHeap(), 0, (x))
+
+dns_t* getWindowsDefDnsServers(void) {
+    FIXED_INFO *pFixedInfo;
+    ULONG ulOutBufLen;
+    DWORD dwRetVal;
+
+    dns_t* result = (dns_t*) MALLOC(sizeof(dns_t));
+    if (result == NULL) return (dns_t*) ERROR_NOT_ENOUGH_MEMORY;
+
+    result->dnsAddresses = (char*) MALLOC(128 * sizeof(char));
```

Comment:
This is never freed. It's not freed in this C code and it is not freed in the Haskell code.